### PR TITLE
Modernize the cookbook

### PR DESCRIPTION
**Pull request by tas50**
The start of a modernization effort. 

- Cookstyle fixes
- Convert Vagrantfile testing to to Test Kitchen
- Add Delivery Local Mode config for runing spec/lint
- Add Github Actions for kicking off Delivery Local Mode on PRs
- Add all the modern boilerplate
- Convert the LWRPs to Custom Resources
- Remove Windows 2008 R2 support since that's EOL
- Improve logging
- Add proper name properties to each resource
- Removes the dep on the windows cookbook by requiring Chef Infra Client 14+
- Switch the resources to use powershell_out instead of the powershell_script resource so that we can get nice clean log output

### CODE REVIEW

**Comment by derekgroh** [comment]
This looks good, do you think `Policyfiles` and `test kitchen` should be introduced as part of the modernization?

**Comment by tas50** [comment]
I was going to get things into Test Kitchen. I'm less worried about Berks since it's just used to depsolve the cookbooks that are needed for TK.

**Comment by tas50** [comment]
@derekgroh pretty sure this is ready at this point

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    - 'test/fixtures/test_cookbooks/test_windows_dhcp/recipes/data_bag.rb'
+
+Lint/ParenthesesAsGroupedExpression:
+  Exclude:
+    - 'test/fixtures/test_cookbooks/test_windows_ad/recipes/ad_objects.rb'
+    - 'test/fixtures/test_cookbooks/test_windows_ad/recipes/setup_dc.rb'
+
+Lint/UselessAssignment:
+  Exclude:
+    - 'Vagrantfile'
```
Comment (derekgroh) [inline .rubocop.yml]: This can be removed

File: README.md [hunk 2]
```diff
@@ -1,25 +1,25 @@
-windows_dhcp Cookbook
-=====================
+# windows_dhcp Cookbook
 [![Join the chat at https://gitter.im/TAMUArch/cookbook.windows_dhcp](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/TAMUArch/cookbook.windows_dhcp?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
```
Comment (derekgroh) [inline README.md:2]: Not really part of modernize, but this isn't used or monitored so I think it should be removed as well.

File: metadata.rb [hunk 3]
```diff
@@ -3,7 +3,6 @@
 maintainer_email 'dgroh@arch.tamu.edu'
 license          'MIT'
 description      'Installs/Configures windows_dhcp'
-long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
 version          '0.2.0'
-supports         'windows', ">= 6.1"
-depends          'windows'
\ No newline at end of file
+supports         'windows', '>= 6.2'
+chef_version     '> 14'
```
Comment (derekgroh) [inline metadata.rb:8]: `issues_url` and `source_url` should be added

**Review by derekgroh — CHANGES_REQUESTED** [review]
The data bag should be moved to the test folder as it isn't a core component of this resource cookbook and serves as an example.

A MIT `LICENSE` file should also be included to resolve all foodcritic warnings.

File: metadata.rb (hunk 3 above)
Comment (tas50) [reply metadata.rb:8]: Fixed

File: README.md (hunk 2 above)
Comment (tas50) [reply README.md:2]: Fixed

File: .rubocop.yml (hunk 1 above)
Comment (tas50) [reply .rubocop.yml]: Fixed

**Comment by derekgroh** [comment]
LGTM

**Comment by tas50** [comment]
Closing due to lack of response

**Pull request closed without merging**